velox/type/tests/VariantTest.cpp

```diff
@@ -285,3 +286,131 @@ TEST(VariantTest, opaqueSerializationNotRegistered) {
       std::make_shared<opaqueSerializationTestStruct>());
   EXPECT_THROW(opaqueAfterRegistration.serialize(), VeloxException);
 }
+
+TEST(VariantTest, toJsonRow) {
+  auto rowType = ROW({{"c0", DECIMAL(20, 3)}});
+  EXPECT_EQ(
+      "[123456.789]",
+      variant::row({static_cast<int128_t>(123456789)}).toJson(rowType));
+
+  rowType = ROW({{"c0", DECIMAL(10, 2)}});
+  EXPECT_EQ(
+      "[12345.67]",
+      variant::row({static_cast<int64_t>(1234567)}).toJson(rowType));
+
+  rowType = ROW({{"c0", DECIMAL(20, 1)}, {"c1", BOOLEAN()}, {"c3", VARCHAR()}});
+  EXPECT_EQ(
+      "[1234567890.1,true,\"test works fine\"]",
+      variant::row({static_cast<int128_t>(12345678901),
+                    variant((bool)true),
+                    variant((std::string) "test works fine")})
+          .toJson(rowType));
+
+  // Row variant negative tests
+  rowType = ROW({{"c0", DECIMAL(10, 3)}});
+  VELOX_ASSERT_THROW(
+      variant::row({static_cast<int128_t>(123456789)}).toJson(rowType),
+      "Wrong type in variant::toJson");
+
+  rowType = ROW({{"c0", DECIMAL(20, 3)}});
+  VELOX_ASSERT_THROW(
+      variant::row({static_cast<int64_t>(123456789)}).toJson(rowType),
+      "Wrong type in variant::toJson");
+  VELOX_ASSERT_THROW(
+      variant::row(
+          {static_cast<int128_t>(123456789),
+           variant((
+               std::
+                   string) "test confirms variant child count is greater than expected"),
+           variant((bool)false)})
+          .toJson(rowType),
+      "Wrong number of fields in a struct in variant::toJson");
+
+  rowType =
+      ROW({{"c0", DECIMAL(19, 4)}, {"c1", VARCHAR()}, {"c3", DECIMAL(10, 3)}});
+  VELOX_ASSERT_THROW(
+      variant::row(
+          {static_cast<int128_t>(12345678912),
+           variant((
+               std::
+                   string) "test confirms variant child count is lesser than expected")})
+          .toJson(rowType),
+      "Wrong number of fields in a struct in variant::toJson");
+
+  // null variant test
+  EXPECT_EQ(
+      "[null,null,null]",
+      variant::row({variant::null(TypeKind::HUGEINT),
+                    variant::null(TypeKind::VARCHAR),
+                    variant::null(TypeKind::BIGINT)})
+          .toJson(rowType));
+}
+
+TEST(VariantTest, toJsonArray) {
+  auto arrayType = ARRAY(DECIMAL(9, 2));
+  EXPECT_EQ(
+      "[1234567.89,6345654.64,2345452.78]",
+      variant::array({static_cast<int64_t>(123456789),
+                      static_cast<int64_t>(634565464),
+                      static_cast<int64_t>(234545278)})
+          .toJson(arrayType));
+
+  arrayType = ARRAY(DECIMAL(20, 3));
+  EXPECT_EQ(
+      "[123456.789,634565.464,234545.278]",
+      variant::array({static_cast<int128_t>(123456789),
+                      static_cast<int128_t>(634565464),
+                      static_cast<int128_t>(234545278)})
+          .toJson(arrayType));
+
+  // negative tests
+  VELOX_ASSERT_THROW(
```

Comment:
Remove this test as this is being handled by another PR.

Reply:
Fixed.